src/app/static/src/app/store/filteredData/getters.ts

```diff
@@ -87,119 +84,52 @@ export const getters = {
 
             const areaId = row.area_id;
 
-            //TODO: This will change when we have a metadata endpoint telling us which column to use as value for each
-            //input data type and indicator
-            //We will also have to deal will potential multiple values per row
-            let indicator: string = "";
-            let valueColumn: string = "";
-            switch (state.selectedDataType) {
-                case (DataType.Survey):
-                    indicator = row["indicator"];
-                    valueColumn = "est";
-                    break;
-                case (DataType.Program):
-                    indicator = "artcov";
-                    valueColumn = "current_art";
-                    break;
-                case (DataType.ANC):
-                    //TODO: once using metadata, we need to allow multiple values per row, as in the case on ANC data
-                    //which contains both prevalence and art data in each row (wide format), unlike survey, which
-                    //is int long format, with an indicator column to show which indicator the value each row provides
-                    indicator = "prev";
-                    valueColumn = "prevalence";
-                    break;
-                case (DataType.Output):
-                    //TODO: output data doesn't currently conform to plotting metadata - use that when it does
-                    indicator = "prev";
-                    valueColumn = "mean";
-                    break;
-            }
-
-            const value = row[valueColumn];
-
             if (!result[areaId]) {
                 result[areaId] = {};
             }
 
-            const indicators = result[areaId];
-            switch (indicator) {
-                case("prev"):
-                    indicators.prev = {value: value, color: ""};
+            const indicatorsForArea = result[areaId];
+            indicatorMetadata.map((i: Indicator) => {
+                const value = row[i.value_column];
```

Comment:
I'm sure this logic isn't right because I still don't understand what value_column, indicator_column etc mean, but you get the idea!

Reply:
Ok, great to hear we're converging! I think making the schema fully agnostic would be nice because the whole motivation for this approach was that the plots should be able to change without our code ever changing - of course if we just treat it as a dictionary without knowledge of the keys that would also work, although then it feels like the schema is unnecessarily opinionated/complicated.